m_npc and friends (4 non-matching)

Kelebek1 opened the pull request:
func_800A8868_jp https://decomp.me/scratch/4F3c8
func_800AADE8_jp https://decomp.me/scratch/heHhr
func_800ADD20_jp https://decomp.me/scratch/97YLU
func_800AE1B8_jp https://decomp.me/scratch/dmRIt

AngheloAlf commented:
Btw I feel like this PR could be easily be 3 PRs, each one for each file. It would be easier to work with and review that way

Kelebek1 commented on src/code/m_npc.c:
```diff
@@ -0,0 +1,4052 @@
+    0x0058, 0x0058, 0x0058, 0x0058, 0x0058, 0x0058, 0x0058, 0x0013, 0x00A5, 0x0000,
+};
+
+static u8 D_8010AF58_jp[] = {
```
? It's not referenced outside the file, so what's wrong with it being static?

Kelebek1 commented on src/code/m_npc_walk.c:
```diff
@@ -0,0 +1,690 @@
+s32 l_goal_block[mNpcW_GOAL_BLOCK_NUM][2];
+s32 l_arrive_stay_count[ANIMAL_NUM_MAX];
+
+static u8 l_girl_time_12[] = { mNpcW_GOAL_SHRINE, mNpcW_GOAL_HOME, mNpcW_GOAL_ALONE };
+static u8 l_girl_time_18_30[] = { mNpcW_GOAL_SHRINE, mNpcW_GOAL_HOME };
```
But they're not referenced anywhere outside this file, and the file is matching.

fig02 replied:
when a symbol is static it gets removed from the map file which is annoying to deal with for decomping + anyone using the codebase in the future for modding for example. we do the same thing in zelda and only use static if necessary.

Kelebek1 commented:
> Btw I feel like this PR could be easily be 3 PRs, each one for each file. It would be easier to work with and review that way

They all rely on eachother, for function names and structs, it would be awkward to do separately.

Kelebek1 commented on src/code/m_npc.c:
I have no idea what their names are.

Prakxo replied:
Check the map names

Kelebek1 replied:
I don't know where this magic map is. I asked before but no one responded.

Kelebek1 commented on include/6B8A70.h:
```diff
@@ -3,20 +3,24 @@
-// void func_80094EE0_jp();
-// void mLd_CopyLandName();
+s32 func_80094E38_jp(u8* dst, u8* src);
+s32 func_80094E90_jp(union memuni_u* dst, u16 dst_id, union memuni_u* src, u16 src_id);
```
There's definitely a land name and ID there, what do you want me to call it?

emilybrooks replied:
I guess just name it unk_XX

fig02 commented on src/code/m_npc_walk.c:
```diff
@@ -0,0 +1,690 @@
+                x = gate->ut1 & 0xF;
+                z = gate->ut1 >> 4;
+
+/* BUG: this is likely meant to be 'if (direction == mRF_DIRECT_WEST || direction == mRF_DIRECT_EAST)' */
+#ifndef BUGFIXES
+                if (direction == mRF_DIRECT_WEST || direction == mRF_DIRECT_WEST) {
+#else
+                if (direction == mRF_DIRECT_WEST || direction == mRF_DIRECT_EAST) {
+#endif
```
+1 on this not being appropriate for the decomp repo. feel free to comment on the bug tho (using the `//! @bug` notation from zelda)

Kelebek1 commented on include/m_font.h:
```diff
@@ -0,0 +1,453 @@
+#define CHAR_ORDINAL_e 219
+#define CHAR_ORDINAL_er 220
+#define CHAR_ORDINAL_re 221
+#define CHAR_BACKSLASH 222
+/* Unused characters */
+#define CHAR_223 223
+#define CHAR_224 224
+#define CHAR_225 225
+#define CHAR_226 226
+#define CHAR_227 227
+#define CHAR_228 228
+#define CHAR_229 229
+#define CHAR_230 230
+#define CHAR_231 231
+#define CHAR_232 232
+#define CHAR_233 233
+#define CHAR_234 234
+#define CHAR_235 235
+#define CHAR_236 236
+#define CHAR_237 237
+#define CHAR_238 238
+#define CHAR_239 239
+#define CHAR_240 240
+#define CHAR_241 241
+#define CHAR_242 242
+#define CHAR_243 243
+#define CHAR_244 244
+#define CHAR_245 245
+#define CHAR_246 246
+#define CHAR_247 247
+#define CHAR_248 248
+#define CHAR_249 249
+#define CHAR_250 250
+#define CHAR_251 251
+#define CHAR_252 252
+#define CHAR_253 253
+#define CHAR_254 254
+#define CHAR_255 255
+
+#define TOTAL_CHARS 256
```
No, but it doesn't make much sense for them to change the characters. All the ones in func_800A83F0_jp make sense at least.

Prakxo replied:
This game doesn't do latin encoding, so this is def wrong for the latin letters

BluRose posted the encoding of each afaik https://github.com/BluRosie/doubutsu-no-mori-binary/blob/default/src/include/chartable.tbl

AngheloAlf commented:
Maide if you aren't sure something is the same between the gcn version and the n64 then don't bring them over to this repo. A lot of stuff changed between versions, for example each are decompiling different languages of the game or the stuff to interact with the GBA doesn't exist in the N64 version.
We also want to properly check members and structs are the same between versions, so don't just name them because other version of the game has it named (either gcn, oot or mm).
Bringing over stuff without checking will just be very confusing in the long run. Please undo the stuff from the headers and files that are not needed and just leave stuff as unk_ for whatever you haven't confirmed

AngheloAlf commented on src/code/m_npc.c:
```diff
@@ -0,0 +1,4055 @@
+    Animal_c* sp30 = mNpc_GetInAnimalP();
+    u8 sp2F = common_data.save.remove_animal_idx;
+
+    gfxprint_color(print, 0xC8, 0x96, 0x32, 0xFF);
```
use dec for color instead of hex

AngheloAlf commented on src/code/m_npc_walk.c:
```diff
@@ -0,0 +1,683 @@
+    s32 res = FALSE;
+    s32 direction;
+
+    *target_ut_x = UT_X_NUM;
+    *target_ut_z = UT_Z_NUM;
```
What does this even mean? What is a UT?

Prakxo replied:
Each acre is known as a block, in a block there's units that are referred as UT, so the specific x and z value are the units in the block

AngheloAlf replied:
And the units are different for x axis and z axis?

Prakxo replied:
Yeah x and z are different

Kelebek1 commented on src/code/m_npc.c:
```diff
@@ -0,0 +1,4055 @@
+    s32 i;
+
+    for (i = 0; i < ANIMAL_NUM_MAX; i++, animal++) {
+        if ((i != (u32)&common_data.save.remove_animal_idx) && (mNpc_CheckFreeAnimalInfo(animal) == FALSE)) {
```
It's not fake, it's what the code does. It's a bug rather than fake.

AngheloAlf replied:
And that case it needs to be labelled as bug then

Kelebek1 commented on include/6BFE60.h:
```diff
@@ -4,15 +4,15 @@
 #include "ultra64.h"
 #include "unk.h"
 
-typedef struct mMl {
-    /* 0x00 */ UNK_TYPE1 unk_00[0xA4];
-} mMl; // size = 0xA4
```
This is Mail which this PR fills in.

Kelebek1 commented on include/m_font.h:
What do you want me to do with it though?

AngheloAlf replied:
Don't bring this defines and just leave the values as hex. You could leave a comment in each use saying what each value means

Kelebek1 commented on src/code/m_npc.c:
```diff
@@ -0,0 +1,4055 @@
+            }
+            var_s4++;
+        }
+        j = 0;
```
Yeah.

Kelebek1 commented on src/code/m_npc.c:
```diff
@@ -0,0 +1,4055 @@
+    0x0027, 0x0027, 0x0027, 0x0027, 0x000A, 0x000A, 0x000A, 0x000A, 0x0058, 0x0058,
+    0x0058, 0x0058, 0x0058, 0x0058, 0x0058, 0x0058, 0x0058, 0x0013, 0x00A5, 0x0000,
+};
+
+u8 D_8010AF58_jp[] = {
+    0x02, 0x05, 0x00, 0x00, 0x01, 0x04, 0x01, 0x05, 0x03, 0x05, 0x05, 0x04, 0x01, 0x00, 0x00, 0x01, 0x04, 0x05, 0x03,
+    0x04, 0x02, 0x02, 0x04, 0x01, 0x05, 0x04, 0x01, 0x03, 0x01, 0x02, 0x01, 0x04, 0x02, 0x05, 0x00, 0x03, 0x04, 0x04,
+    0x02, 0x03, 0x00, 0x03, 0x02, 0x00, 0x03, 0x01, 0x05, 0x04, 0x04, 0x05, 0x03, 0x00, 0x02, 0x00, 0x03, 0x01, 0x01,
+    0x05, 0x00, 0x01, 0x02, 0x03, 0x00, 0x02, 0x00, 0x00, 0x02, 0x05, 0x01, 0x04, 0x03, 0x01, 0x02, 0x03, 0x02, 0x01,
+    0x02, 0x02, 0x03, 0x00, 0x03, 0x04, 0x03, 0x01, 0x00, 0x02, 0x02, 0x04, 0x03, 0x02, 0x00, 0x03, 0x00, 0x03, 0x04,
+    0x02, 0x05, 0x05, 0x04, 0x03, 0x04, 0x02, 0x01, 0x00, 0x02, 0x01, 0x00, 0x03, 0x05, 0x00, 0x03, 0x00, 0x05, 0x01,
+    0x00, 0x05, 0x05, 0x00, 0x04, 0x02, 0x05, 0x01, 0x02, 0x03, 0x03, 0x03, 0x01, 0x00, 0x02, 0x01, 0x04, 0x01, 0x04,
+    0x05, 0x04, 0x00, 0x03, 0x01, 0x05, 0x04, 0x04, 0x03, 0x05, 0x03, 0x03, 0x00, 0x04, 0x02, 0x02, 0x05, 0x05, 0x03,
+    0x01, 0x02, 0x02, 0x04, 0x05, 0x03, 0x00, 0x03, 0x04, 0x01, 0x02, 0x04, 0x00, 0x01, 0x01, 0x03, 0x03, 0x02, 0x03,
+    0x04, 0x05, 0x00, 0x02, 0x00, 0x01, 0x05, 0x04, 0x04, 0x05, 0x01, 0x03, 0x04, 0x01, 0x04, 0x05, 0x04, 0x02, 0x05,
+    0x04, 0x02, 0x04, 0x02, 0x03, 0x05, 0x02, 0x00, 0x05, 0x04, 0x01, 0x05, 0x00, 0x00, 0x05, 0x05, 0x01, 0x00, 0x05,
+    0x04, 0x00, 0x01, 0x00, 0x05, 0x03, 0x01, 0x01, 0x01, 0x00, 0x00,
+};
+
+u32 D_8010B034_jp[] = {
+    0x11111111, 0x11111111, 0x11111111, 0x11111111, 0x11111111, 0x11111111, 0x11111111, 0x11111111,
+    0x11111111, 0x11111111, 0x11111111, 0x11111111, 0x11111111, 0x11111111, 0x11111111, 0x11111111,
+};
+
+u32 D_8010B074_jp[] = {
+    0x09090909,
+};
+
+u32 D_8010B078_jp[] = {
+    0x13131313, 0x13131313, 0x13131313, 0x13131313, 0x13131313, 0x13131313, 0x13131313, 0x13131313, 0x13131313,
+};
+
+u32 D_8010B09C_jp[] = {
+    0x08080808, 0x08080808, 0x08080808, 0x08080808, 0x08080808, 0x08080808, 0x08080808, 0x08080808,
+};
+
+u32 D_8010B0BC_jp[] = {
+    0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C,
+    0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C,
+    0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C, 0x0C0C0C0C,
```
m_npc uses D_8010AF58_jp, that's why I put it all in here.

Kelebek1 commented on src/code/m_npc.c:
```diff
@@ -0,0 +1,4055 @@
+    s32 res = FALSE;
+
+    if ((animal->home_info.block_x == 0xFF) && (animal->home_info.block_z == 0xFF) &&
+        (ACTOR_FGNAME_GET_F000(animal->id.npc_id) != FGNAME_F000_E)) {
```
fgname is an npc id, so, maybe?

Kelebek1 commented on src/code/m_npc.c:
```diff
@@ -0,0 +1,4055 @@
+}
+
+AnmPersonalID_c* func_800A7DA0_jp(AnmPersonalID_c* ids, s32 num_ids, s32 arg0, s32 arg1, s32 arg2) {
+    UNUSED s32 pad;
+    AnmPersonalID_c* sp68;
+    u8 sp67;
+    UNUSED s32 pad1;
+    s32 sp5C;
+    u16 sp5A;
+    s32 j;
+    s32 i;
+    UNUSED s32 pad2[2];
+    s32 sp44;
+    Animal_c* var_s4;
+    s32 var_fp;
+    s32 var_s2;
+
+    var_s4 = common_data.save.animals;
+    sp68 = NULL;
+    var_s2 = 0;
+    sp5C = num_ids;
+    sp67 = common_data.save.now_npc_max;
+    sp5A = 0xFFFF;
```
Can't be done due to stack.

Prakxo commented on src/code/m_npc.c:
```diff
@@ -0,0 +1,4055 @@
+
+u16 D_8010B908_jp[] = { 0, 0xA012, 0xFFFF, 0xFFFF, 0xFFFF, 0xFFFF, 0xFFFF, 0xFFFF, 0xFFFF };
+
+s32 D_8010B91C_jp[] = { 4, 4, 2, 2, 3, 4 };
```
```suggestion
s32 spec_table[] = { 4, 4, 2, 2, 3, 4 };
```

Prakxo commented on src/code/m_npc.c:
```diff
@@ -0,0 +1,4055 @@
+
+s32 D_8010B91C_jp[] = { 4, 4, 2, 2, 3, 4 };
+
+s32 D_8010B934_jp[] = { 0x28, 0x32, 0x3C, 0x46, 0x50, 0x5A, 0x64 };
```
```suggestion
s32 npc_grow_table[] = { 0x28, 0x32, 0x3C, 0x46, 0x50, 0x5A, 0x64 };
```

Prakxo commented on src/code/m_npc.c:
```diff
@@ -0,0 +1,4055 @@
+    /* 0x03 */ u8 unk_03;
+} Struct_D_8010B950_jp; // size = 0x4
+
+Struct_D_8010B950_jp D_8010B950_jp[] = {
```
```suggestion
Struct_D_8010B950_jp l_npc_temper[] = {
```

Prakxo reviewed (commented):
I reanamed the data variables that are the same in contents in GC, I didn't bother with bss since I don't know its contents, nor I want to investigate a lot

emilybrooks reviewed (commented):
game_play is the original name for the variable and shouldn't be renamed.

Kelebek1 commented:
> game_play is the original name for the variable and shouldn't be renamed.

Shouldn't matter if the repo has decided camelcase. If you knew the devs used snake case and chose a different style anyway, should stick to it. Why randomly mix them on a whim?

AngheloAlf commented:
Part of the reasoning of mixing the two styles to make clear as possible what's an original name and what are names we came up with

AngheloAlf commented on include/m_common_data.h:
```diff
@@ -125,78 +128,102 @@ typedef struct FamicomEmuCommonData {
     /* 0x10142 */ u8 unk10142[0x7];
     /* 0x10149 */ u8 unk_10149;
     /* 0x1014A */ u8 unk_1014A;
-    /* 0x1014B */ u8 unk_1014B; // named "wipeType" in AC GCN decomp
```
Dude, don't remove comments like this

Kelebek1 commented on yamls/jp/code.yaml:
```diff
@@ -246,27 +251,31 @@
-      - { start: 0x75B600, type: bss, vram: 0x80137960, name: code/80137960 }
+      - { start: 0x73F4D0, type: .bss, vram: 0x80126EA0, name: code/m_common_data }
+      - { start: 0x73F4D0, type:  bss, vram: 0x80137960, name: code/80137960 }
       - { start: 0x73F4D0, type: .bss, vram: 0x80138E50, name: code/m_debug}
-      - { start: 0x73F4D0, type: bss, vram: 0x80138E60, name: code/m_debug_display}
+      - { start: 0x73F4D0, type:  bss, vram: 0x80138E60, name: code/m_debug_display}
       - { start: 0x73F4D0, type: .bss, vram: 0x80141E10, name: code/m_lights }
-      - { start: 0x75B600, type: .bss, vram: 0x80141FA0, name: code/m_malloc }
-      - { start: 0x75B600, type: bss, vram: 0x80141FE0, name: code/80141FE0 }
+      - { start: 0x73F4D0, type: .bss, vram: 0x80141FA0, name: code/m_malloc }
+      - { start: 0x73F4D0, type:  bss, vram: 0x80141FE0, name: code/80141FE0 }
+      - { start: 0x73F4D0, type: .bss, vram: 0x80142C00, name: code/m_npc }
+      - { start: 0x73F4D0, type: .bss, vram: 0x801438B0, name: code/m_npc_walk }
       - { start: 0x73F4D0, type: .bss, vram: 0x80143900, name: code/m_player_call }
       - { start: 0x73F4D0, type: bss, vram: 0x80143910, name: code/80143910 }
       - { start: 0x73F4D0, type: .bss, vram: 0x80144670, name: code/m_submenu }
       - { start: 0x73F4D0, type: bss, vram: 0x80144690, name: code/80144690 }
       - { start: 0x73F4D0, type: .bss, vram: 0x80145020, name: code/game }
-      - { start: 0x73F4D0, type: bss, vram: 0x80145060, name: code/80145060 }
+      - { start: 0x73F4D0, type:  bss, vram: 0x80145060, name: code/80145060 }
       - { start: 0x73F4D0, type: .bss, vram: 0x80145390, name: code/irqmgr }
-      - { start: 0x73F4D0, type: bss, vram: 0x80145620, name: code/80145620}
+      - { start: 0x73F4D0, type:  bss, vram: 0x80145620, name: code/80145620}
       - { start: 0x73F4D0, type: .bss, vram: 0x801462B0, name: code/sys_matrix }
-      - { start: 0x73F4D0, type: bss, vram: 0x801462C0, name: code/801462C0 }
+      - { start: 0x73F4D0, type:  bss, vram: 0x801462C0, name: code/801462C0 }
       - { start: 0x73F4D0, type: .bss, vram: 0x801492C0, name: code/zurumode }
-      - { start: 0x73F4D0, type: bss, vram: 0x801492D0, name: code/801492D0 }
+      - { start: 0x73F4D0, type:  bss, vram: 0x801492D0, name: code/801492D0 }
```
What toher entries? The bss is just a block on its own, since it needs to be kept ordered this helps to see the addresses, otherwise it's hard to compare them.

Kelebek1 commented:
> Did you really decided to rename even unrelated stuff to this PR because they weren't following the naming style? By doing that you are creating even more work to the reviewers, which was something we already complained to you

Besides game_play I don't think so? I didn't have a specific list of what I changed and what I didn't, I went through all the files that I added/modified and changed those, which did include common_data. If the style is going to be a random mishmash, then you either need to make a specific list of what everything needs to be named, or just ignore style and change it later as you want.

emilybrooks commented:
pr is being split up into smaller prs
